Restrict maximum number of I2C bytes send in one transmission.

t-paul opened the pull request:
I had the issue that some of the LEDs did not light as bright as the others when using the Arduino library, but the same effect did not happen with the Raspberry Pi sample code.

After some digging, I think the cause is a limitation of the Wire library. While the AS1130 datasheet states an unlimited number of data bytes per I2C transmission the Wire library uses an internal buffer of 32 bytes (`BUFFER_LENGTH` in `Wire.h`).

The patch changes the `fillMemory()` method to split transfers which fixes the issue for me on a Leonardo board.

I've tested with the following code added to the LED-Test example:
```
void loop() {
  for (int a = 0;a < 10;a++) {
    ledDriver.setBlinkAndPwmSetAll(0, 0, 10);
    delay(200);
    ledDriver.setBlinkAndPwmSetAll(0, 0, 255);
    delay(200);
  }
  delay(1000);
}
```

t-paul committed:
Restrict maximum number of I2C bytes send in one transmission.

LuckyResistor commented:
Great you found this problem, and thank you for the detailed report and patch! I did not expect this. I will investigate the issue and integrate this patch. Using an Arduino Zero, I had no such problems. I will compare the used Wire libraries.

LuckyResistor reviewed (changes requested):
The `write` function returns the number of written bytes. In case of the one byte version, either `0` or `1`. To make your patch more universal, could you just check this return value. As soon a call of write returns `0` the transmission of one block could be started.

It will make the code slightly more complex, but it would work with any defined buffer size in the Wire library. It also eleminates the need for the constant `cMaxWireBytes`.

t-paul commented:
Right, makes sense. I'll have a look...

t-paul committed:
Change fillMemory() to split transfers when the internal Wire buffer is full.

This relies on the return value of Wire.write() which will return 0 in
case the buffer is full.

t-paul commented:
Updated the pull request to use the `Wire.write()` return value.

Now there's the, probably theoretical case, that this turns into an endless loop if always only the address byte transfer succeeds but never any data byte is sent. I'm not sure if it's worth checking for that...

Some test cases using RAM-Configuration 2 (*is there some way to unlock that via software?*):
```
using namespace lr;
AS1130 ledDriver;

void setup() {
  Wire.begin();
  delay(100); 
  ledDriver.setRamConfiguration(AS1130::RamConfiguration2);
  ledDriver.setOnOffFrameAllOn(1, 1);
  ledDriver.setOnOffFrameAllOn(0, 0);
  ledDriver.setCurrentSource(AS1130::Current30mA);
  ledDriver.setScanLimit(AS1130::ScanLimitFull);
  ledDriver.startPicture(0);
  ledDriver.startChip();
}

void loop() {
  for (uint8_t a = 0;a <= 110;a++) {
    uint8_t l = 25 * abs(10 - a % 20);
    ledDriver.setBlinkAndPwmSetAll(0, 0, l);
    delay(20);
  }
  delay(1000);

  for (int a = 11;a < 256;a++) {
    uint8_t i = abs(a % 22 - 11);
    ledDriver.fillMemory(AS1130::RS_BlinkAndPwmSet, 0x18, 0, 132);
    ledDriver.fillMemory(AS1130::RS_BlinkAndPwmSet, 0x18 + 11 * i, 255 - a, 10);
    delay(10);
  }
  delay(1000);

  for (int a = 6;a < 133;a++) {
    uint8_t p = a % 2;
    uint8_t i = abs(a % 12 - 6);
    ledDriver.setBlinkAndPwmSetAll(p, 0, 0);
    ledDriver.fillMemory(AS1130::RS_BlinkAndPwmSet + p, 0x18 + 66 - 11 * i, 40 * i, 22 * i);
    delay(10);
    ledDriver.startPicture(p);
  }
  delay(1000);
}
```

LuckyResistor reviewed (approved):
Thank you very much!
